tests/httpwrappers/tests.py

```diff
@@ -299,6 +300,18 @@ def test_fromkeys_noniterable(self):
             QueryDict.fromkeys(0)
 
 
+class HttpResponseBaseTests(SimpleTestCase):
+    def test_charset_setter(self):
+        r = HttpResponseBase()
+        r.charset = "utf-8"
+        self.assertEqual(r.charset, HttpResponseBase(charset="utf-8").charset)
```

Comment:
Do we need new objects? It think it's enough to check a value:
```suggestion
        self.assertEqual(r.charset, "utf-8")
```

Reply:
My original thoughts was that a setter might not simply assign a value onto it, so I just try to check the consistency between the setter and the constructor(`__init__()`). I'm not sure if it's correct, is it better to just check a value?

Reply:
felix's test is better because it tests the value of `charset` is correct.

You can get a false positive with `assert r.charset == HttpResponseBase(charset='utf-8').charset` you're just asserting that the attribute's values are the same without asserting what the actual value is.

Reply:
Additionally… setters and getters are separate features and would ideally be tested in separate tests. Regressions are easier to pinpoint when your test->feature mapping is more direct.

Reply:
Okay, thanks for the explanation.